CRM/Core/WorkflowMessage/UFNotify.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/*
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC. All rights reserved.                        |
+ |                                                                    |
+ | This work is published under the GNU AGPLv3 license with some      |
+ | permitted exceptions and without any warranty. For full license    |
+ | and copyright information, see https://civicrm.org/licensing       |
+ +--------------------------------------------------------------------+
+ */
+
+use Civi\WorkflowMessage\GenericWorkflowMessage;
+
+/**
+ * Receipt sent when someone receives a copy of profile that has been filled out.
+ *
+ * @method int getProfileID()
+ * @method $this setProfileID(int $profileID)
+ * @method array getProfileFields()
+ * @method $this setProfileFields(array $profileFields)
+ *
+ * @support template-only
+ *
+ * @see CRM_Core_BAO_UFGroup::commonSendMail
+ */
+class CRM_Core_WorkflowMessage_UFNotify extends GenericWorkflowMessage {
+  public const WORKFLOW = 'uf_notify';
+
+  /**
+   * @var int
+   *
+   * @scope tplParams
+   */
+  protected $profileID;
+
+  /**
+   * @var array
+   *
+   * @scope tplParams as values
+   */
+  protected $profileFields;
+
+  /**
+   * @var string
+   *
+   * @scope tplParams
+   */
+  protected $contactLink;
+
+  public function getContactLink(): string {
+    return CRM_Utils_System::url('civicrm/contact/view',
+      "reset=1&cid=" . $this->getContactID(),
+      TRUE, NULL, FALSE, FALSE, TRUE
+    );
+  }
+
+  /**
+   * @var string
+   *
+   * @scope tplParams as grouptitle
+   */
+  protected $groupTitle;
+
+  public function getGroupTitle(): string {
+    return CRM_Core_DAO::getFieldValue('CRM_Core_DAO_UFGroup', $this->getProfileID(), 'frontend_title');
```

Comment:
I got error because my profile had missing frontend_title

`TypeError: CRM_Core_WorkflowMessage_UFNotify::getGroupTitle(): Return value must be of type string, null returned in CRM_Core_WorkflowMessage_UFNotify->getGroupTitle() (line 65 of /Users/pradeep/Sites/drupal7/sites/all/modules/civicrm/CRM/Core/WorkflowMessage/UFNotify.php).
`

<br class="Apple-interchange-newline">

Reply:
@pradpnayak did you test this by backporting it - in 5.72 frontend_title should be a required field

Reply:
Hi @eileenmcnaughton 

My bad, tested by upgrading to 5.71 beta. The frontend title is not updated with the title field and the field is required through UI.